Allow std::array as operator argument and return

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34399 Allow std::array as operator argument and return**
* #34335 Allow ArrayRef as kernel parameter

Custom ops can now take std::array as arguments and return it.
This PR also moves the ops in native_functions.yaml that were blocked by this to now `use_c10_dispatcher: full`.

Differential Revision: [D20315072](https://our.internmc.facebook.com/intern/diff/D20315072/)

smessmer committed:
Allow std::array as operator argument and return

Custom ops can now take std::array as arguments and return it.
This PR also moves the ops in native_functions.yaml that were blocked by this to now `use_c10_dispatcher: full`.

Differential Revision: [D20315072](https://our.internmc.facebook.com/intern/diff/D20315072/)

[ghstack-poisoned]

smessmer committed:
Update on "Allow std::array as operator argument and return"


Custom ops can now take std::array as arguments and return it.
This PR also moves the ops in native_functions.yaml that were blocked by this to now `use_c10_dispatcher: full`.

Differential Revision: [D20315072](https://our.internmc.facebook.com/intern/diff/D20315072/)

[ghstack-poisoned]

ezyang reviewed (approved):
It's kind of copy pastey with the list code, but I'm not sure if there's a good way to eliminate it.

smessmer committed:
Update on "Allow std::array as operator argument and return"


Custom ops can now take std::array as arguments and return it.
This PR also moves the ops in native_functions.yaml that were blocked by this to now `use_c10_dispatcher: full`.

Differential Revision: [D20315072](https://our.internmc.facebook.com/intern/diff/D20315072/)

[ghstack-poisoned]

smessmer committed:
Update on "Allow std::array as operator argument and return"


Custom ops can now take std::array as arguments and return it.
This PR also moves the ops in native_functions.yaml that were blocked by this to now `use_c10_dispatcher: full`.

Differential Revision: [D20315072](https://our.internmc.facebook.com/intern/diff/D20315072/)

[ghstack-poisoned]
